docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,308 @@
+ */
+DecimalNumber: (DecimalDigits | (DecimalDigits? '.' DecimalDigits)) ([eE] '-'? DecimalDigits)?;
+//@doc:inline
+fragment DecimalDigits: [0-9] ('_'? [0-9])* ;
+
+
+/**
+ * An identifier in solidity has to start with a letter, a dollar-sign or an underscore and
+ * may additionally contain numbers after the first symbol.
+ */
+Identifier: IdentifierStart IdentifierPart*;
+//@doc:inline
+fragment IdentifierStart: [a-zA-Z$_];
+//@doc:inline
+fragment IdentifierPart: [a-zA-Z0-9$_];
+
+WS: [ \t\r\n\u000C]+ -> skip ;
+COMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+LINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN);
+
+mode AssemblyBlockMode;
+
+//@doc:inline
+AssemblyDialect: '"evmasm"';
+AssemblyLBrace: '{' -> popMode, pushMode(YulMode);
+
+AssemblyBlockWS: [ \t\r\n\u000C]+ -> skip ;
+AssemblyBlockCOMMENT: '/*' .*? '*/' -> channel(HIDDEN) ;
+AssemblyBlockLINE_COMMENT: '//' ~[\r\n]* -> channel(HIDDEN) ;
+
+mode YulMode;
+
+YulBreak: 'break';
+YulCase: 'case';
+YulContinue: 'continue';
+YulDefault: 'default';
+YulFalse: 'false';
+YulFor: 'for';
+YulFunction: 'function';
+YulIf: 'if';
+YulLeave: 'leave';
+YulLet: 'let';
+YulSwitch: 'switch';
+YulTrue: 'true';
+
+/**
+ * Builtin functions in the EVM Yul dialect.
+ */
+YulEVMBuiltin:
+    'stop' | 'add' | 'sub' | 'mul' | 'div' | 'sdiv' | 'mod' | 'smod' | 'exp' | 'not'
+    | 'lt' | 'gt' | 'slt' | 'sgt' | 'eq' | 'iszero' | 'and' | 'or' | 'xor' | 'byte'
+    | 'shl' | 'shr' | 'sar' | 'addmod' | 'mulmod' | 'signextend' | 'keccak256'
+    | 'pop' | 'mload' | 'mstore' | 'mstore8' | 'sload' | 'sstore' | 'msize' | 'gas'
+    | 'address' | 'balance' | 'selfbalance' | 'caller' | 'callvalue' | 'calldataload'
+    | 'calldatasize' | 'calldatacopy' | 'extcodesize' | 'extcodecopy' | 'returndatasize'
+    | 'returndatacopy' | 'extcodehash' | 'create' | 'create2' | 'call' | 'callcode'
+    | 'delegatecall' | 'staticcall' | 'return' | 'revert' | 'selfdestruct' | 'invalid'
+    | 'log0' | 'log1' | 'log2' | 'log3' | 'log4' | 'chainid' | 'origin' | 'gasprice'
+    | 'blockhash' | 'coinbase' | 'timestamp' | 'number' | 'difficulty' | 'gaslimit';
+
+YulLBrace: '{' -> pushMode(YulMode);
+YulRBrace: '}' -> popMode;
+YulLParen: '(';
+YulRParen: ')';
+YulAssign: ':=';
+YulPeriod: '.';
+YulComma: ',';
+YulArrow: '->';
+
+/**
+ * Yul identifiers consist of letters, dollar signs, underscores and numbers, but may not start with a number.
+ */
+YulIdentifier: YulIdentifierStart YulIdentifierPart*;
+//@doc:inline
+fragment YulIdentifierStart: [a-zA-Z$_];
+//@doc:inline
+fragment YulIdentifierPart: [a-zA-Z0-9$_];
+/**
+ * Hex literals in Yul consist of a prefix and one or more hexadecimal digits.
+ */
+YulHexNumber: '0' 'x' [0-9a-fA-F]+;
+/**
+ * Decimal literals in Yul may be zero or any sequence of decimal digits without leading zeroes.
+ */
+YulDecimalNumber: '0' | ([1-9] [0-9]*);
+/**
+ * String literals in Yul consist of one or more double quoted strings that may contain escape sequences
```

Comment:
Where are single quotes disallowed?

Reply:
Maybe they aren't... but then we're lacking test cases for that :-). Would be trivial to add them. I just checked, they indeed seem to be supported.

Reply:
Just allowed single quoted strings, but restricted to printable characters.